### ISSUE TITLE
Spectral Input gets stuck during generation

### ISSUE THREAD

**Issue by elggem** [ibab/tensorflow-wavenet#250]
I'm attempting to feed WaveNet a distribution of scalars instead of one_hot encoded amplitudes (could be seen as spectral information of waves, but the data is something else with dependencies over time and between values). Ive done some modifications to the network like modifying the output activations and I'm getting closer to modelling the data. I have to do further experimentation regarding the receptive field size as of yet.

One issue I am facing in some trails is that WaveNet is getting stuck during generation. The output either collapses into a repetitive and high frequency pattern or is completely uniform. See attached some trail runs where this is happening (x axis shows the different values and y axis is time):

![3999](https://cloud.githubusercontent.com/assets/6736746/24641699/4a5ee95a-1934-11e7-8849-e0627b87ba06.png)
![3500](https://cloud.githubusercontent.com/assets/6736746/24641700/4a624532-1934-11e7-908f-79ab2b5da1b9.png)

Has someone encountered this during training on waveforms or is this a problem exclusive to the modifications I am doing? Is it purely related to insufficient training or is there another cause that I can look at? Any pointers much appreciated.

**Comment by veqtor** [comment]
I've seen other have issues with wavenet hanging after generating a few samples.
Out of curiosity, the output, is it linear or sigmoid activations?

**Comment by elggem** [comment]
In the latest runs ive changed it to sigmoid which yields better results generally for my particular data. However both exhibit the hanging issue.

**Comment by elggem** [comment]
I've added in noise during generation to try to shake it out of the repetitive pattern its generating, but apart from the added noise there is no difference, it still gets stuck in the one state... Also after a lot more steps of  learning it tends to get stuck less often....

**Comment by elggem** [comment]
Hmm by now I think it definately hangs more when using a sigmoid activation on the output layer!

**Comment by lemonzi** [comment]
Sigmoid activations can exhibit [vanishing gradients](https://www.quora.com/What-is-the-vanishing-gradient-problem), so a linear output should be more robust if you manage to get good results with it. I would also try to replace the `relu` activations inside the model with something more gradient-friendly, like `elu` or a leaky `relu`. You could also try adding more regularization to the weights and biases of the inner layers.

**Comment by elggem** [comment]
We've managed to get the model to work by increasing the filter width and also by training it more. Increasing the number of residual and dilation channels also seems to help convergence! If anyone is interested in applying WaveNet to these kinds of data hit me up.
